pkg/autodiscovery/listeners/docker.go

```diff
@@ -180,9 +180,15 @@ func (l *DockerListener) processEvent(e *docker.ContainerEvent) {
 	l.m.RUnlock()
 
 	if found {
-		if e.Action == "die" {
+		switch e.Action {
+		case "die":
 			l.removeService(cID)
-		} else {
+		case "start":
+			// Container restarted with the same ID within 5 seconds.
```

Comment:
is `docker restart` the only way we could receive this event?

Reply:
That I can think of, yes. But even if it's not the only one, this should yield this behavior:
"stop: wait for 5 seconds before removing completely"
"start of the same container within the 5 seconds: wait 5 seconds before starting a new service"